aimon/decorators/evaluate.py

```diff
@@ -301,6 +301,14 @@ def evaluate(
         if "instructions" in record and "instruction_adherence" in config:
             # Only pass instructions if instruction_adherence is specified in the config
             payload["instructions"] = record["instructions"] or ""
+        
+        if "retrieval_relevance" in config and len(config)==1:
```

Comment:
task_definition is required if retrieval_relevance is ever specified in the config.

Currently this only requires it when retrieval_relevance is the only detector in the config